polaris-react/src/components/Stack/Stack.scss

```diff
@@ -1,42 +1,14 @@
 @import '../../styles/common';
 
 .Stack {
+  @include responsive-props('stack', 'gap', 'gap');
   // stylelint-disable -- Polaris component custom properties
   --pc-stack-align: initial;
   --pc-stack-order: initial
-  --pc-stack-gap-xs: 0;
-  --pc-stack-gap-sm: var(--pc-stack-gap-xs);
-  --pc-stack-gap-md: var(--pc-stack-gap-sm);
-  --pc-stack-gap-lg: var(--pc-stack-gap-md);
-  --pc-stack-gap-xl: var(--pc-stack-gap-lg);
```

Comment:
I think we need to keep these initial values in, because the `responsive-props` mixin sets them to `initial` and it's causing UI regressions with spacing. This will probably also resolve all of the chromatic diffs.